connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/streaming/StreamingQueryManager.scala

```diff
@@ -0,0 +1,147 @@
+    val response = executeManagerCmd(_.setGet(id))
+    if (response.hasQuery) {
+      RemoteStreamingQuery.fromStreamingQueryInstanceResponse(sparkSession, response.getQuery)
+    } else {
+      null
+    }
+  }
+
+  /**
+   * Wait until any of the queries on the associated SQLContext has terminated since the creation
+   * of the context, or since `resetTerminated()` was called. If any query was terminated with an
+   * exception, then the exception will be thrown.
+   *
+   * If a query has terminated, then subsequent calls to `awaitAnyTermination()` will either
+   * return immediately (if the query was terminated by `query.stop()`), or throw the exception
+   * immediately (if the query was terminated with exception). Use `resetTerminated()` to clear
+   * past terminations and wait for new terminations.
+   *
+   * For correctly documenting exceptions across multiple queries, users need to stop all of them
+   * after any of them terminates with exception, and then check the `query.exception()` for each
+   * query.
+   *
+   * @since 3.5.0
+   */
+  // TODO(SPARK-43299): verity the behavior of this method after JVM client-side error-handling
+  // framework is supported and modify the doc accordingly.
+  def awaitAnyTermination(): Unit = {
+    executeManagerCmd(_.getAwaitAnyTerminationBuilder.build())
+  }
+
+  /**
+   * Wait until any of the queries on the associated SQLContext has terminated since the creation
+   * of the context, or since `resetTerminated()` was called. Returns whether any query has
+   * terminated or not (multiple may have terminated). If any query has terminated with an
+   * exception, then the exception will be thrown.
+   *
+   * If a query has terminated, then subsequent calls to `awaitAnyTermination()` will either
+   * return `true` immediately (if the query was terminated by `query.stop()`), or throw the
+   * exception immediately (if the query was terminated with exception). Use `resetTerminated()`
+   * to clear past terminations and wait for new terminations.
+   *
+   * For correctly documenting exceptions across multiple queries, users need to stop all of them
+   * after any of them terminates with exception, and then check the `query.exception()` for each
+   * query.
+   *
+   * @since 3.5.0
+   */
+  // TODO(SPARK-43299): verity the behavior of this method after JVM client-side error-handling
+  // framework is supported and modify the doc accordingly.
+  def awaitAnyTermination(timeoutMs: Long): Boolean = {
+    require(timeoutMs > 0, "Timeout has to be positive")
+    executeManagerCmd(
+      _.getAwaitAnyTerminationBuilder.setTimeoutMs(
+        timeoutMs)).getAwaitAnyTermination.getTerminated
+  }
+
+  /**
+   * Forget about past terminated queries so that `awaitAnyTermination()` can be used again to
+   * wait for new terminations.
+   *
+   * @since 3.5.0
+   */
+  def resetTerminated(): Unit = {
+    executeManagerCmd(_.setResetTerminated(true))
+  }
+
+  private def executeManagerCmd(
```

Comment:
Not a blocker: this method looks very similar to `executeQueryCmd` in `StreamingQuery`. Do you think if there's an easy way to abstract that logic out as a common/util method that could be shared among different commands? If no easy solution is available, you can ignore this comment.

Reply:
I think this is possible but not necessarily a urgent need as we only have two of these right now. Also these methods :`getStreamingQueryManagerCommandBuilder` and `getStreamingQueryCommandBuilder` and other similar methods makes it a little bit to generally I'd say...

Reply:
OK, sg.